sfdoc/publish/utils.py

```diff
@@ -63,20 +77,41 @@ def find_bundle_root_directory(origpath):
 
     # look down from origpath away from /
     for root, dirs, files in os.walk(origpath):
-        print(root, files)
         if matchfile in files:
             return root
 
     raise FileNotFoundError("Cannot find log.txt to identify root directory!")
 
 
-def s3_key_to_relative_pathname(key):
-    build_id, branch, path = key.split("/", 2)
-    assert int(build_id) + 1
-    assert branch in ("draft", "release")
-    return path
-
-
 def bundle_relative_path(bundle_root, path):
+    """Remove the bundle part of the path"""
     assert os.path.isabs(path)
     return os.path.relpath(path, bundle_root)
+
+
+logger = logging.getLogger("commands")
+
+
+def info(*args):
+    logger.info(*args)
+
+
+def run_command(*args, **kwargs):
+    info(" ".join(args))
+    return subprocess.run(args, **kwargs, check=True, text=True)
+
+
+def s3_sync(source, target, delete=False):
+    """Sync directory to S3 or S3 to directory"""
+    args = ["aws", "s3", "sync", source, target]
+    
+    if delete:
+        args.append("--delete")
+    run_command(*args)
+
+
+def sync_directories(source, target):
+    """Keep two directories in sync"""
+    if not source.endswith("/"):
+        source += "/"
+    run_command("rsync", "-r", source, target)
```

Comment:
Is it available on heroku?

Reply:
rsync  version 3.1.1  protocol version 31

Reply:
Okay, no real problem then. It does seem odd to take so much care about path separators and then depend on a unix-y utility. But it doesn't make sense to reinvent rsync.